core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -939,7 +939,31 @@ private CompiledQuery alterTable(SqlExecutionContext executionContext) throws Sq
                         tok = expectToken(lexer, "'add index' or 'cache' or 'nocache'");
                         if (SqlKeywords.isAddKeyword(tok)) {
                             expectKeyword(lexer, "index");
-                            alterTableColumnAddIndex(tableNamePosition, columnNameNamePosition, columnName, writer);
+                            tok = SqlUtil.fetchNext(lexer);
+                            int indexValueCapacity = -1;
+                           
+                            if ( tok != null ){
+                                if (!SqlKeywords.isCapacityKeyword(tok) ){
+                                    throw SqlException.$(lexer.lastTokenPosition(), "'capacity' expected");
+                                }
+                                else {
+                                    tok = expectToken(lexer, "capacity value");
+                                    try {
+                                        indexValueCapacity = Numbers.parseInt(tok);
+                                        if ( indexValueCapacity <= 0 ){
+                                            throw SqlException.$(lexer.lastTokenPosition(), "positive integer literal expected as index capacity");
+                                        }
+                                        if (indexValueCapacity != Numbers.ceilPow2(indexValueCapacity) ){
+                                            throw SqlException.$(lexer.lastTokenPosition(), "power of 2 expected as index capacity");
```

Comment:
This is inconsistent with `create table` statement. The following works:

```sql
CREATE TABLE x(a symbol index CAPACITY 129);
```

when you select the following:

```sql
table_columns('x');
```

you should see that capacity is rounded to 256.

on other hand the following doesn't work:

```sql
CREATE TABLE x(a symbol);

alter table x alter column a add index capacity 129;
```

I think both `create table` and `alter table` should work consistently. I also feel that the SQL impl should automatically round number to power of two. Otherwise requiring user to provide accurate number is a clunky user experience.

Reply:
I agree. Fixed in latest commit .